libs/domain/product/images/src/images.component.ts

```diff
@@ -1,248 +1,186 @@
 @hydratable('mouseover')
-export class ProductImagesComponent extends ProductComponentMixin<ProductImagesComponentOptions>() {
-  static styles = styles;
-
-  protected product$ = new ProductController(this)
-    .getProduct()
-    .pipe((product) => this.reset(product));
-
-  protected options$ = new ContentController(this)
-    .getOptions()
-    .pipe((options) => this.reset(options));
-
-  protected active$ = new BehaviorSubject(0);
-
-  protected override render(): TemplateResult {
-    return html`
-      ${asyncValue(
-        combineLatest([this.product$, this.options$, this.active$]),
-        ([product, options, active]) => {
-          const media = this.resolveImages(product, options);
-          return this.wrap(
-            options,
-            this.renderMain(options, media, active),
-            this.renderNavigation(options, media, active)
-          );
-        }
-      )}
-    `;
-  }
+export class ProductImagesComponent extends ProductMixin(
+  ContentMixin<ProductImagesComponentOptions>(LitElement)
+) {
+  static styles = productImageStyles;
 
-  protected resolveImages(
-    product: Product | null,
-    options: Partial<ProductImagesComponentOptions>
-  ): ProductMedia[] {
-    const set = !options.mediaSet
-      ? product?.mediaSet?.[0]
-      : product?.mediaSet?.find((set) => set.name === options.mediaSet);
-    return set?.media ?? [];
+  @state() set active(value: number) {
+    this.scrollToImage(value);
   }
 
-  /**
-   * Wrapper element that reflects layout options. The DOM order of the main
-   * image and the navigation are _adapted_ based on the configured layout.
-   * The order of the preview and navigation as well as the reflected options
-   * are crucial for proper styling.
-   *
-   * We cannot set those on the host element, since there's no support for `style.setAttribute`
-   * and `toggleAttribute` on the host in lit ssr.
-   */
-  protected wrap(
-    options: ProductImagesComponentOptions,
-    main: TemplateResult,
-    navigation: TemplateResult
-  ): TemplateResult {
-    const cssVar = (prop: string, value?: number): string | undefined =>
-      value ? `${prop}:${value}${isNaN(value) ? '' : 'px'};` : '';
-
-    let style = '';
-    style += cssVar('--product-images-main-height', options.mainImageHeight);
-    style += cssVar('--product-images-main-width', options.mainImageWidth);
-    style += cssVar('--product-images-thumb-height', options.thumbHeight);
-    style += cssVar('--product-images-thumb-width', options.thumbWidth);
-    if (options.gridItemsPerColumn) {
-      style += `--oryx-grid-items-per-column: ${options.gridItemsPerColumn}`;
+  protected override render(): TemplateResult | void {
+    if (this.product) {
+      const media = this.resolveImages();
+      const main = this.renderMainLayout(media);
+      const navigation = this.renderNavigationLayout(media);
+
+      return html`
+        <oryx-layout
+          navigation=${this.componentOptions?.navigationPosition ||
```

Comment:
You have default options, why do you need fallback values then?

Reply:
The problem is currently with EB. If you clear a value, it creates an empty string rather than an undefined, which is why we need to use `||` operator.

Reply:
But in this case should it use the defaultOptions? I believe we will have at least something, EB options or default ones, but still